packages/@react-spectrum/searchwithin/src/SearchWithin.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, SlotProvider, useFocusableRef, useResizeObserver, useStyleProps} from '@react-spectrum/utils';
+import {DOMProps, FocusableRef, SpectrumLabelableProps, StyleProps} from '@react-types/shared';
+import {Field} from '@react-spectrum/label';
+import React, {ReactNode, useCallback, useLayoutEffect, useRef, useState} from 'react';
+import styles from '@adobe/spectrum-css-temp/components/searchwithin/vars.css';
```

Comment:
we'll need to add inputgroup css so that focus rings extend around both the field and button, see https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/combobox/src/ComboBox.tsx#L46

Reply:
Per team discussion and review of the spec, we're going to let the search field and picker button have their own focus rings. We do still need to fix styles so that neither inside side of the rings is overlapped by the other element.